### PR TITLE
Update app-custom-theme.md

### PR DESCRIPTION
**Pull request by padupe**
Signed-off-by: Paulo Peixoto <72894267+padupe@users.noreply.github.com>

## Hey, I just made a Pull Request!
Added documentation for using custom icons.
After study and validation at @madeiramadeirabr.

Co-Authored by: @accezar, @ericksposito @Monnylly

image

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by awanlin** [comment]
Just a heads up, we are going to have some conflict between our PRs as was working on the same thing here: https://github.com/backstage/backstage/pull/12696

File: docs/getting-started/app-custom-theme.md [hunk 1]
```diff
@@ -273,6 +273,115 @@ const LogoFull = () => {
...
+In the `assets` directory, you can create the `customIcons.tsx` file.
+
+```tsx
+// customIcons.tsx
+import React from 'react';
+import ExampleSVG from './icons/example.svg';
+
+export const ExampleIcon = () => {
+  return <img alt="" src={ExampleSVG} />;
+};
```
Comment (awanlin) [inline docs/getting-started/app-custom-theme.md:349]: With this method your icon won't be theme ware and may not look good in light or dark themes. If you us `SvgIcon` from `@material-ui/core` you'll get that support automatically. Here's an example: https://github.com/backstage/backstage/blob/master/plugins/azure-devops/src/components/AzurePipelinesIcon/AzurePipelinesIcon.tsx

File: docs/getting-started/app-custom-theme.md [hunk 2]
```diff
@@ -273,6 +273,115 @@ const LogoFull = () => {
 };
 ```
 
+## Custom Icons
+
+You can also customize the Project's _default_ icons.
+
+We can change the following icons:
```
Comment (awanlin) [inline docs/getting-started/app-custom-theme.md:280]: I would just make this a link to the code where we have the default icons that way it stays up to date: https://github.com/backstage/backstage/blob/master/packages/app-defaults/src/defaults/icons.tsx

**Review by awanlin — COMMENTED** [review]
This is awesome, thanks for taking the time to put this together. Just a pair of comments

File: docs/getting-started/app-custom-theme.md (hunk 2 above)
Comment (padupe) [reply docs/getting-started/app-custom-theme.md:280]: Perfect!

File: docs/getting-started/app-custom-theme.md (hunk 1 above)
Comment (padupe) [reply docs/getting-started/app-custom-theme.md:349]: Excellent suggestion!
Thanks for sharing!

File: docs/getting-started/app-custom-theme.md [hunk 3]
```diff
@@ -273,6 +273,77 @@ const LogoFull = () => {
 };
 ```
 
+## Custom Icons
+
+You can also customize the Project's _default_ icons.
+
+We can change the following [icons](https://github.com/backstage/backstage/blob/master/packages/app-defaults/src/defaults/icons.tsx).
```
Comment (jhaals) [inline docs/getting-started/app-custom-theme.md:280]:
```suggestion
You can change the following [icons](https://github.com/backstage/backstage/blob/master/packages/app-defaults/src/defaults/icons.tsx).
```

File: docs/getting-started/app-custom-theme.md [hunk 4]
```diff
@@ -273,6 +273,77 @@ const LogoFull = () => {
...
+);
+```
+
+### Using the custom icon
+
+In your front-end application, locate the `src/components` folder.
+
+You will find the `App.tsx` file.
+
+The `app` constant, instantiates the `createApp` function, in this constant we will use the "icons" parameter that receives an object.
```
Comment (jhaals) [inline docs/getting-started/app-custom-theme.md:320]:
```suggestion
Supply your custom icon in `packages/app/src/App.tsx`
```

File: docs/getting-started/app-custom-theme.md [hunk 5]
```diff
@@ -273,6 +273,77 @@ const LogoFull = () => {
...
+We can change the following [icons](https://github.com/backstage/backstage/blob/master/packages/app-defaults/src/defaults/icons.tsx).
+
+### Requirements
+
+- Files in `.svg` format;
+- React components created for the icons
+
+### Create React Component
+
+In your front-end application, locate the `src` folder. We suggest creating the `assets/icons` directory.
```
Comment (jhaals) [inline docs/getting-started/app-custom-theme.md:289]: Perhaps this example can be slimmed down and just suggesting the full path to the `ExampleIcon` and then referring to the code example in https://github.com/backstage/backstage/blob/master/plugins/azure-devops/src/components/AzurePipelinesIcon/AzurePipelinesIcon.tsx ?

File: docs/getting-started/app-custom-theme.md [hunk 6]
```diff
@@ -273,6 +273,77 @@ const LogoFull = () => {
...
 
+## Custom Icons
+
+You can also customize the Project's _default_ icons.
+
+We can change the following [icons](https://github.com/backstage/backstage/blob/master/packages/app-defaults/src/defaults/icons.tsx).
+
+### Requirements
+
+- Files in `.svg` format;
```
Comment (jhaals) [inline docs/getting-started/app-custom-theme.md:284]:
```suggestion
- Files in `.svg` format
```

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the contribution!

I've added some suggestions.
CI is currently failing because the file need to be correctly formatted, you can do this with `yarn prettier -w docs/getting-started/app-custom-theme.md`

File: docs/getting-started/app-custom-theme.md [hunk 7]
```diff
@@ -273,6 +273,71 @@ const LogoFull = () => {
...
+
+> Another example [here](https://github.com/backstage/backstage/blob/master/plugins/azure-devops/src/components/AzurePipelinesIcon/AzurePipelinesIcon.tsx), if you want to ensure proper behavior in light and dark themes.
+
+```tsx
+// customIcons.tsx
+import { SvgIcon, SvgIconProps } from '@material-ui/core';
+
+import React from 'react';
+
+export const HeaderIcon = (props: SvgIconProps) => (
```
Comment (jhaals) [inline docs/getting-started/app-custom-theme.md:299]:
```suggestion
export const ExampleIcon = (props: SvgIconProps) => (
```

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks! I found one minor thing and then I think it's good to go :)

File: docs/getting-started/app-custom-theme.md (hunk 7 above)
Comment (jhaals) [reply docs/getting-started/app-custom-theme.md:299]: This should match the example below where we import it so it should probably be ExampleIcon here :)

Comment (padupe) [reply docs/getting-started/app-custom-theme.md]: Wow, I forgot to update this "guy" haha
Thanks for the review!

**Comment by padupe** [comment]
Thanks for suggestions @jhaals!

**Review by jhaals — APPROVED** [review]
Thank you! 👍

**Comment by suuus** [comment]
Congrats on your first PR merged @padupe! Happy to count you as a contributor.

**Comment by padupe** [comment]
Thanks @suuus 🚀

**Comment by DanielMcFarlin** [comment]
Hey there still seems to be a small gap in this documentation, well at least from my understanding. I see how I can create a custom icon and import it into `app.tsx` file and add it with the `createApp()` logic. But how do I refer to that icon within a newly created plugin? Is there any docs on that? I could use some help in that area.

**Pull request merged**